### PR TITLE
[IR] Unify max alignment for arguments with generic max align.

### PR DESCRIPTION
**Pull request by efriedma-quic**
The 2^14 limit was completely arbitrary; the generic limit is still arbitrary, but at least it's the same arbitrary limit as everything else.

While I'm here, also add a verifier check for the ByValOrByRefSize.

### CODE REVIEW

File: llvm/lib/IR/Verifier.cpp [hunk 1]
```diff
@@ -2021,31 +2014,43 @@ void Verifier::verifyParameterAttrs(AttributeSet Attrs, Type *Ty,
...
-        Align MaxAlign(ParamMaxAlignment);
-        Check(AttrAlign <= MaxAlign,
-              "Attribute 'align' exceed the max size 2^14", V);
-      }
       SmallPtrSet<Type *, 4> Visited;
       Check(Attrs.getByValType()->isSized(&Visited),
             "Attribute 'byval' does not support unsized types!", V);
+      Check(DL.getTypeAllocSize(Attrs.getByValType()).getKnownMinValue() <
+                (1ULL << 32),
+            "huge 'byval' arguments are unsupported", V);
```
Comment (nikic) [inline llvm/lib/IR/Verifier.cpp:2028]: Tests for the new checks?

**Comment by Fznamznon** [comment]
Thank you! That helps with the problem I was originally trying to solve with https://github.com/llvm/llvm-project/pull/98629 . Does it make sense to keep https://github.com/llvm/llvm-project/pull/98629 but with the increased limit?

**Comment by phoebewang** [comment]
> Thank you! That helps with the problem I was originally trying to solve with #98629 . Does it make sense to keep #98629 but with the increased limit?

Given #98629 has side effect on Darwin targets, I'd like to not pursuit it if the original problem getting solve.

**Review by phoebewang — APPROVED** [review]
LGTM.

**Pull request merged**